apps/studio/components/interfaces/SignIn/SignInForm.tsx

```diff
@@ -52,8 +52,8 @@ const SignInForm = () => {
 
         toast.success(`Signed in successfully!`, { id: toastId })
         await queryClient.resetQueries()
-
-        router.push(getReturnToPath())
+        const returnTo = getReturnToPath()
+        router.push(returnTo !== '/sign-in' ? returnTo : '/projects')
```

Comment:
I think this is a little easier to read the other way round 😅

```suggestion
        // since we're already on the /sign-in page, prevent redirect loops
        router.push(returnTo === '/sign-in' ? '/projects' : returnTo)
```